bin/proving-service/src/proxy/update_workers.rs

```diff
@@ -0,0 +1,155 @@
+            Ok(res) => {
+                if !res {
+                    error!("Failed to read request header");
+                    create_response_with_error_message(
+                        &mut http,
+                        "Failed to read request header".to_string(),
+                    )
+                    .await
+                    .ok();
+                    return None;
+                }
+            },
+            Err(e) => {
+                error!("HTTP server fails to read from downstream: {e}");
+                create_response_with_error_message(
+                    &mut http,
+                    format!("HTTP server fails to read from downstream: {e}"),
+                )
+                .await
+                .ok();
+                return None;
+            },
+        }
+
+        info!("Successfully get a new request to update workers");
+
+        // Extract and parse query parameters, if there are not any, return early.
+        let query_params = match http.req_header().as_ref().uri.query() {
+            Some(params) => params,
+            None => {
+                let error_message = "No query parameters provided".to_string();
+                error!("{}", error_message);
+                create_response_with_error_message(&mut http, error_message).await.ok();
+                return None;
+            },
+        };
+
+        let update_workers: Result<UpdateWorkers, _> = serde_qs::from_str(query_params);
+        let update_workers = match update_workers {
+            Ok(workers) => workers,
+            Err(err) => {
+                let error_message = format!("Failed to parse query parameters: {}", err);
+                error!("{}", error_message);
+                create_response_with_error_message(&mut http, error_message).await.ok();
+                return None;
+            },
+        };
+
+        // Update workers and handle potential errors
+        if let Err(err) = self.lb_state.update_workers(update_workers).await {
+            let error_message = format!("Failed to update workers: {}", err);
+            error!("{}", error_message);
+            create_response_with_error_message(&mut http, error_message).await.ok();
+            return None;
+        }
+
+        create_workers_updated_response(&mut http, self.lb_state.num_workers().await)
+            .await
+            .ok();
+
+        info!("Successfully updated workers");
+
+        None
+    }
+
+    /// Provide HTTP server options used to override default behavior. This function will be called
+    /// every time a new connection is processed.
+    fn server_options(&self) -> Option<&HttpServerOptions> {
+        Some(&self.server_opts)
+    }
+}
+
+// HELPERS
+// ================================================================================================
+
+/// Create a 200 response for updated workers
+///
+/// It will set the X-Worker-Count header to the number of workers.
+async fn create_workers_updated_response(
```

Comment:
We don't need to emit a new metric at this point for `WORKER_COUNT` because we expect to update that only on the health checks, correct?

Reply:
We do emit a WORKER_COUNT metric when updating the workers through this method (health check does it too). THe logic is placed in the `LoadBalancerState::update_workers` method, which is called internally during the processing of each `update_worker` request.